Automattic/wp-calypso
# Remove Flux from lib/infinite-list scroll positions store

Replaces the Flux store in `lib/infinite-list` with a very simple implementation using `Map` and getter and (throttled) setter functions.

The store only has a getter/setter API, and no change events that anyone would subscribe to. That's why it doesn't even need Redux.

**How to test:**
- enable debug logging by enabling it in console and reloading:
```js
localStorage.debug = 'calypso:infinite-list:scroll-store'
```
- go to a route that uses `InfiniteList`, e.g., the media library
- scroll down in the list of media
- go to another page
- click the "Back" button and navigate back to media library
- check that when navigating using browser history, the scroll position is reconstructed to the previous value
- while testing, watch the debug messages in console and verify there's nothing unexpected going on

@blowery added some Reader-related functionality to `InfiniteList` in the past. He could know some more sophisticated testing steps, especially for the Reader Stream usage 🙂 

Fixes #24178.
